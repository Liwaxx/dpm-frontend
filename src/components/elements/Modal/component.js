import React from 'react';
import Button from '@material-ui/core/Button';
import { TableRow, TableCell } from '@material-ui/core';
import Modal from '@material-ui/core/Modal';
import PropTypes from 'prop-types';
import { IMAGES } from '../../../configs';

function getModalStyle() {
  const top = 50;
  const left = 50;

  return {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
  };
}

export default class Component extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
      open: false,
      query: '',
    };
  }

  handleIncrement() {
    if (this.state.count == 11) {
      this.setState({ count: 0 });
    } else {
      this.setState({
        count: this.state.count + 1
      });
    }
  }

  handleDecrement() {
    if (this.state.count == 0) {
      this.setState({ count: 11 });
    } else {
      this.setState({
        count: this.state.count - 1
      });
    }
  }

  handleOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  _renderWeek() {
    const dataArray = ['Januari', 'Februari', 'Maret', 'April', 'Mei', 'Juni',
      'Juli', 'Agustus', 'September', 'Oktober', 'November', 'Desember'];
    const { classes } = this.props;

    return (
      <TableRow className={classes.line}>
        <TableCell>
          <button id="modal-decrement" onClick={() => this.handleDecrement()} className={classes.left}>
            <img src={IMAGES.ARROW_LEFT} className={classes.closeImage} />
          </button>
        </TableCell>
        <TableCell>
          <div className={classes.tahun}>
            {dataArray[this.state.count]}
          </div>
        </TableCell>
        <TableCell>
          <button id="modal-increment" onClick={() => this.handleIncrement()} className={classes.right}>
            <img src={IMAGES.ARROW_RIGHT} />
          </button>
        </TableCell>
      </TableRow>
    );
  }

  render() {
    const { classes } = this.props;
    return (
      <div className={classes.Modal}>
        <Button id="modal-open" onClick={this.handleOpen} className={classes.detail}>Oktober, Minggu 2</Button>
        <Modal open={this.state.open}>
          <div style={getModalStyle()} className={classes.paper}>
            <div className={classes.header}>
              <div className={classes.week}>
                {this._renderWeek()}
              </div>
              <div className={classes.close}>
                <Button id="modal-close" onClick={this.handleClose}>
                  <img src={IMAGES.CLOSE} />
                </Button>
              </div>
            </div>
            <div className={classes.line1}>
              <Button > Minggu ke 1 </Button>
              <Button > Minggu ke 2 </Button>
            </div>
            <div className={classes.line2}>
              <Button > Minggu ke 3 </Button>
              <Button > Minggu ke 4 </Button>
            </div>
          </div>
        </Modal>
      </div>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object.isRequired,
};