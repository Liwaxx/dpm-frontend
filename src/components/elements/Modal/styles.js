const styles = theme => ({
  paper: {
    position: 'absolute',
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    width: '300px',
    height: '207px',
    marginLeft: '-15%',
    marginTop: '-10%',
  },
  container: {
    backgroundColor: '#FFF',
    fontFamily: 'roboto',
    fontSize: '20px',
  },
  detail: {
    color: '#53B9EC',
  },  
  header: {
    marginLeft: '8px',
  },
  close: {
    marginLeft: '240px',
    marginTop: '-50px',
    width: '20px',
    height: '20px',
  },
  closeImage: {
    width: '50px',
  },
  line1: {
    marginLeft: '40px',
    marginTop: '40px',
  },
  line2: {
    marginLeft: '40px',
    marginTop: '20px',
  },
  left: {
    border: 'none',
    backgroundColor: '#fff',
    marginLeft: '20px',
  },
  right: {
    border: 'none',
    backgroundColor: '#fff',
    marginLeft: '-70px',
  },
  line: {
    marginLeft: '110px',
  },
  tahun: {
    fontFamily: 'roboto',
    marginLeft: '-70px',
    fontSize: '20px',
  },
});
export default styles;