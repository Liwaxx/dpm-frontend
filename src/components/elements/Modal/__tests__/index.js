import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Modal from '../index';
import Component from '../component';

jest.mock('../../../../configs');

describe('Modal', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Modal />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('increases state.count when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true });
    wrapper.find('#modal-increment').first().simulate('click');
    expect(wrapper.state('count')).toEqual(1);
  });

  it('returns state.count to 0 when the button clicked and count is 11', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true, count: 11 });
    wrapper.find('#modal-increment').first().simulate('click');
    expect(wrapper.state('count')).toEqual(0);
  });

  it('decreases state.count when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true, count: 2 });
    wrapper.find('#modal-decrement').first().simulate('click');
    expect(wrapper.state('count')).toEqual(1);
  });

  it('returns state.count to 11 when the button clicked and count is 0', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true, count: 0 });
    wrapper.find('#modal-decrement').first().simulate('click');
    expect(wrapper.state('count')).toEqual(11);
  });

  it('opens modal when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.find('#modal-open').first().simulate('click');
    expect(wrapper.state('open')).toEqual(true);
  });

  it('closes modal when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true });
    wrapper.find('#modal-close').first().simulate('click');
    expect(wrapper.state('open')).toEqual(false);
  });
});