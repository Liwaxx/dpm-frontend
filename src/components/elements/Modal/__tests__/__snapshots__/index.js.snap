// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root-39 MuiButton-root-13 MuiButton-text-15 MuiButton-flat-18 Component-detail-3"
    disabled={false}
    id="modal-open"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-14"
    >
      Oktober, Minggu 2
    </span>
    <span
      className="MuiTouchRipple-root-42"
    />
  </button>
</div>
`;
